libraries/AP_Scripting/examples/rev_thr_on_land.lua

```diff
@@ -0,0 +1,181 @@
+--[[
+    This script automatically engages reverse thrust on planes in FBWA mode on touchdown.
+    Prerequisities:
+    - ArduPlane 4.1 or later
+    - A Fixed Wing platform capable of CTOL operation
+    - ESCs capable of reversing motor direction
+    - A Separate RC channel for triggering throttle reverse, assigned by RCx_FUNCTION=64
+    - A good rangefinder (downward facing) capable of reliably reading [0.1, 5]m of range
+    - A separate RC channel that will serve as an "arming" switch for reverse throttle operation. 
+      This script uses RC_ARM_CHANNEL_ for this purpose (i.e. reverse throttle is armed whenever this switch PWM >1500, 
+      set by RC_ARM_CHANNEL_HI_PWM_)
+    Params to adjust:
+    - THRESH_LAND_HEIGHT_ (m) - this is the height read by rangefinder which corresponds to a plane's height above ground
+      on touchdown. Set this a couple of cms higher than the height reading by your rangefider when the aircraft is on the ground
+      but well within the height that the rangefinder can reliably read.
+    - LAND_DET_TIME_ (s) - time during which rangefinder height estimate is consistently below THRESH_LAND_HEIGHT_
+    - MIN_GROUND_SPEED_ (m/s) - minimum ground speed at which the autopilot will disengage reverse thrust and switch to
+      idle forward thrust
+    - REV_THRUST_HIGH_PWM_, REV_THRUST_IDLE_PWM_ - max and min pwm values for RC_CHANNEL_NUM_ when in reverse throttle mode
+    - REV_THRUST_CHG_PWM_PS_(pwm) - this is the motors' linear acceleration time in seconds, from REV_THRUST_IDLE_PWM_ to REV_THRUST_HIGH_PWM_
+    - MAXTIME_REV_THRUST_APPLIED_MS_ (ms) - maximum time, in seconds, reverse throttle is allowed to apply
+    - TIME_THROT_WAIT_MS_ (ms) - time the throttle is suppressed, sitting at forward idle
+    - REV_THRUST_SPOOL_DPWM_ (pwm) - initial increase in pwm to apply when in reverse throttle mode
+]]--
+
+
+----------***---Parameter Binding (Taken from VTOL-quicktune.lua)**** 
+
+local PARAM_TABLE_KEY = 9
+local PARAM_TABLE_PREFIX = "REV_"
+
+-- bind a parameter to a variable
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+ 
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return bind_param(PARAM_TABLE_PREFIX .. name)
```

Comment:
Move to the creation with name, it does all the asserting for you.
```suggestion
   return Parameter(PARAM_TABLE_PREFIX .. name)
```